adding snowflake behavior changes configs (#6905)
dbt-labs/docs.getdbt.com: website/docs/reference/global-configs/snowflake-changes.md

---
title: "Snowflake adapter behavior changes"
id: "snowflake-changes"
sidebar: "Snowflake"
---

> ## The enable_truthy_nulls_equals_macro flag

Reviewer: ```suggestion
```

> The `enable_truthy_nulls_equals_macro` flag is `False` by default. If this is set to `True` in your dbt_project.yml, this will enable null-safe equality on the dbt equals macro which is utilized in the incremental and snapshot materializations. 

Reviewer: ```suggestion
The `enable_truthy_nulls_equals_macro` flag is `False` by default. Setting it to `True` in your `dbt_project.yml` file enables null-safe equality on the dbt equals macro, which is used in the incremental and snapshot materializations. 
```

> For example, without the flag, when you compare NULL using `=`, it does not return true, even when comparing NULL = NULL. Making it null safe allows for proper comparisons with NULL.

Reviewer: ```suggestion
For example, when you compare NULL using `=` without the flag, it doesn't return `TRUE`, even when comparing `NULL = NULL`. Making it null safe allows for proper comparisons with `NULL`.
```

> If both values are NULL, it evaluates to TRUE instead of UNKNOWN.

Reviewer: ```suggestion
If both values are `NULL`, it evaluates to `TRUE` instead of `UNKNOWN`.
```
